### PR TITLE
account for custom log path

### PR DESCRIPTION
**Pull request by martin-kretzschmar**

### CODE REVIEW

File: uberAgentSupport-macOS/uberAgentSupport.zsh [hunk 1]
```diff
@@ -40,27 +40,56 @@ if [ "$(id -u)" != "0" ]; then
 fi
 
 # Create a structured directory tree inside the bundle path
-mkdir -p "${bundlePath}"/{Config/Local,Config/CCFM,Logs/Daemon,Logs/Users,CrashReports}
+mkdir -p "${bundlePath}"/{Config/Local,Config/CCFM,Logs/Daemon,Logs/Users,Logs/Remote,CrashReports}
```
Comment (dev-sai-collab) [inline uberAgentSupport-macOS/uberAgentSupport.zsh:43]: Optional: add checks before this line to ensure hostname and bundlePath are not empty.

```
if [ -z "${bundlePath}" ]; then
    echo "Error: bundlePath is not set."
    exit 1
fi

if [ -z "${hostname}" ]; then
    echo "Error: hostname is not set."
    exit 1
fi
```

File: uberAgentSupport-macOS/uberAgentSupport.zsh [hunk 2]
```diff
@@ -40,27 +40,56 @@ if [ "$(id -u)" != "0" ]; then
...
+# Redirect all future command outputs (both stdout and stderr) to the specified log file
+exec > >(tee -a $scriptLogPath) 2>&1
+
+# Function to collect log files from either default or specified path
+collectLogs() {
+    local remoteLogfilePath="$1"
+
+    if [ -d "${remoteLogfilePath}" ]; then
+        echo "Collecting logs from ${remoteLogfilePath}..."
+        find "${remoteLogfilePath}" -type f -name "*${hostname}*" -exec cp {} "${bundlePath}/Logs/Remote/" \;
```
Comment (dev-sai-collab) [inline uberAgentSupport-macOS/uberAgentSupport.zsh:57]: Do we need error handling here, i.e., could the find command fail?

File: uberAgentSupport-macOS/uberAgentSupport.zsh [hunk 3]
```diff
@@ -40,27 +40,56 @@ if [ "$(id -u)" != "0" ]; then
...
+    else
+        echo "Log file path ${remoteLogfilePath} is not accessible or does not exist. Collecting logs from default locations."
+        cp -R /Library/Logs/uberAgent/* "${bundlePath}/Logs/Daemon/"
```
Comment (dev-sai-collab) [inline uberAgentSupport-macOS/uberAgentSupport.zsh:60]: Could cp fail here? Do we need error handling?

File: uberAgentSupport-macOS/uberAgentSupport.zsh [hunk 4]
```diff
@@ -40,27 +40,56 @@ if [ "$(id -u)" != "0" ]; then
...
+        
+        # Make sure to copy user-specific logs
+        for dir in /Users/*/
+        do
+            username=$(basename $dir)
```
[the remark this answers is not in the document]
Comment (martin-kretzschmar) [reply uberAgentSupport-macOS/uberAgentSupport.zsh:65]: Adding quotes.

File: uberAgentSupport-macOS/uberAgentSupport.zsh (hunk 1 above)
Comment (martin-kretzschmar) [reply uberAgentSupport-macOS/uberAgentSupport.zsh:43]: Added check for bundlePath.

File: uberAgentSupport-macOS/uberAgentSupport.zsh (hunk 2 above)
Comment (martin-kretzschmar) [reply uberAgentSupport-macOS/uberAgentSupport.zsh:57]: find usually does not fail silently, but of course can fail for various reasons. I think we can leave as is for the time being.

**Review by dev-sai-collab — APPROVED** [review]

**Pull request merged**